src/App.tsx

```diff
@@ -1,110 +1,135 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const options = [3, 5, 10, 20];
+
+interface ItemsListProps {
+  itemL: string[];
+}
+
+interface PaginationProps {
+  pageCount: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+}
+
+export const ItemList: React.FC<ItemsListProps> = ({ itemL }) => (
+  <ul>
+    {itemL.map(item => (
+      <li key={item} data-cy="item">
+        {item}
+      </li>
+    ))}
+  </ul>
+);
+
+export const Pagination: React.FC<PaginationProps> = ({
+  pageCount,
+  currentPage,
+  onPageChange,
```

Comment:
The `Pagination` component should be stateless and use props `total` and `perPage` instead of `pageCount`. Update the component to accept these props and calculate `pageCount` internally based on them.